### PR TITLE
Use --strip-dwarf when emitting source maps

### PR DESCRIPTION
**Pull request by kripken**
This avoids a build with source maps also having full
DWARF debug sections, which are not needed, and can
be extremely large.

@dschuff you mentioned we may also want to not emit
full debug info in object files - are there clang flags for
that? Or some processing we should do?

Depends on https://github.com/emscripten-core/emscripten/pull/9869

### CODE REVIEW

File: tools/shared.py [hunk 1]
```diff
@@ -2908,6 +2908,9 @@ def run_binaryen_command(tool, infile, outfile=None, args=[], debug=False, stdou
       cmd += ['--input-source-map=' + infile + '.map']
       cmd += ['--output-source-map=' + outfile + '.map']
       cmd += ['--output-source-map-url=' + Settings.SOURCE_MAP_BASE + os.path.basename(Settings.WASM_BINARY_FILE) + '.map']
+      # remove any dwarf debug info sections, as the source map has the info,
+      # so they are not needed (and may be very large)
+      cmd += ['--strip-dwarf']
```
Comment (sbc100) [inline tools/shared.py]: Any reason not to just use `--strip-debug` here?

Comment (kripken) [reply tools/shared.py]: It would do too much. In a source maps build, we do want the SourceMapUrl section, which strip-debug would remove.

**Comment by dschuff** [comment]
This makes sense for now (i.e. to strip the DWARF when we emit source maps), regardless of whether the DWARF is just a line table or full debug info.

Clang has a flag `-gline-tables-only` that prevents non-sourcemappable debug info from being generated, and we should probably be emitting that by default for now to avoid wasted build time.
But of course we need a way to test real debug info for now, and eventually we'll have DWARF debug info as a separate thing, so we'll have to think a bit more about how we want to flag that. But that is a different PR

**Comment by RReverser** [comment]
Can we strip DWARF not only when emitting source maps, but on any Wasm transforms as well (until proper DWARF adjustment is implemented)? Right now most of the time this leads to situations where Wasm has been transformed, but DWARF has not, and all code offsets are off but assumed to be valid by debuggers.

**Comment by dschuff** [comment]
That makes sense to do (and as I mentioned above I think we want to rethink our debug info behavior from first principles), but I also think the browser shouldn't be interpreting DWARF yet (other than behind a flag), especially in preference to source maps. If that's what it does, I'd consider that behavior a bug.

**Comment by kripken** [comment]
Thanks @RReverser , done.

**Comment by RReverser** [comment]
> especially in preference to source maps. If that's what it does, I'd consider that behavior a bug.

That's not what it does when source maps are present, but when they're not, then they would fall back to DWARF info emitted by Emscripten, which is currently, unfortunately, invalid. Hence the suggestion to just remove it altogether until we have proper DWARF support.

Thanks @kripken!

**Pull request merged**